src/DotNetLightning.Core/Serialization/LightningStream.fs

```diff
@@ -416,4 +419,9 @@ type LightningReaderStream(inner: Stream) =
             rest <- int32 this.Length - int32 this.Position
         result |> Seq.toArray
          
-        
+    member this.ReadShutdownScriptPubKey(): ShutdownScriptPubKey =
+        let script = this.ReadScript()
+        match ShutdownScriptPubKey.TryFromScript script with
+        | Ok shutdownScript -> shutdownScript
+        | Error errorMsg -> failwith errorMsg
```

Comment:
Should raise `FormatException`

Reply:
Good catch.

Fixed. And I've rebased the PR.